latch_cli/docker_utils/__init__.py

```diff
@@ -0,0 +1,212 @@
+                            mkdir /root/.conda && \\
+                            bash Miniconda3-latest-Linux-x86_64.sh -b -p /opt/conda && \\
+                            rm -f Miniconda3-latest-Linux-x86_64.sh && \\
+                            conda init bash \
+                        """
+                    ).strip(),
+                ],
+                order=DockerCmdBlockOrder.PRECOPY,
+            ),
+            DockerCmdBlock(
+                comment="build and configure conda environment",
+                commands=[
+                    f"copy {conda_env_file} /root/{conda_env_file}",
+                    f"run conda env create -f {conda_env_file} --name {env_name}",
+                    f"""shell ["conda", "run", "-n", "{env_name}", "/bin/bash", "-c"]""",
+                    "run /opt/conda/bin/pip install --upgrade latch",
+                ],
+                order=DockerCmdBlockOrder.PRECOPY,
+            ),
+        ]
+
+    # python package in workflow directory (poetry, distutils, etc.)
+    if (pkg_root / "setup.py").exists() or (pkg_root / "pyproject.toml").exists():
+        print("Add local package to python environment")
+        commands.append(
+            DockerCmdBlock(
+                comment="add local package to python environment",
+                commands=["run python3 -m pip install -e /root/"],
+                order=DockerCmdBlockOrder.POSTCOPY,
+            )
+        )
+
+    # python requirements.txt
+    if (pkg_root / "requirements.txt").exists():
+        print("Add requirements from `requirements.txt` to python environment")
+        commands.append(
+            DockerCmdBlock(
+                comment="add requirements from `requirements.txt` to python environment",
+                commands=[
+                    "copy requirements.txt /root/requirements.txt",
+                    "run python3 -m pip install -r requirements.txt",
+                ],
+                order=DockerCmdBlockOrder.PRECOPY,
+            )
+        )
+
+    return commands
+
+
+def generate_dockerfile(pkg_root: Path, outfile: Path) -> None:
+    """Generate a best effort Dockerfile from files in the workflow directory.
+
+    Args:
+        pkg_root: A path to a workflow directory.
+        outfile: The path to write the generated Dockerfile.
+
+    Example:
+
+        >>> generate_dockerfile(Path("test-workflow"), Path("test-workflow/Dockerfile"))
+            # The resulting file structure will look like
+            #   test-workflow
+            #   ├── Dockerfile
+            #   └── ...
+    """
+
+    print("Generating Dockerfile")
+    try:
+        with open(pkg_root / latch_constants.pkg_config) as f:
+            config: LatchWorkflowConfig = LatchWorkflowConfig.from_json(f.read())
+            print("  - base image:", config.base_image)
+            print("  - latch version:", config.latch_version)
+    except FileNotFoundError:
```

Comment:
```suggestion
    except FileNotFoundError as e:
```
and `raise ... from e`